src/concurrency/Sync.java

```diff
@@ -3,29 +3,61 @@
 import java.util.List;
 import java.util.ArrayList;
 import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
 
 class Sync extends Thread {
-    List nums = new ArrayList<>();
+    static List nums = new ArrayList<>();
 
     Sync(String name) {
         super(name);
     }
 
-    @Override 
+    @Override
     public void run() {
+        System.out.println("running");
         Random rand = new Random();
 
-        for(int i=0; i<100; i++) {
+        for (int i = 0; i < 100; i++) {
             int randInt = rand.nextInt(100);
             this.nums.add(randInt);
         }
     }
 
-    public static void main(String[] args)  {
-        Sync syncThread = new Sync("sync thread");
-        syncThread.start();
+    public static void main(String[] args) {
```

Comment:
when this runs, it looks likely to finish before anything is output - REF: https://www.geeksforgeeks.org/joining-threads-in-java/#:~:text=Thread%20class%20provides%20the%20join,is%20executed%20by%20the%20program.